.ci/apm-beats-update.groovy

```diff
@@ -99,3 +85,53 @@ pipeline {
   //   }
   // }
 }
+
+def beatsUpdate() {
+  def os = "linux"
+  def goRoot = "${env.WORKSPACE}/.gvm/versions/go${GO_VERSION}.${os}.amd64"
+
+  withEnv([
+    "HOME=${env.WORKSPACE}",
+    "GOPATH=${env.WORKSPACE}",
+    "GOROOT=${goRoot}",
+    "PATH=${env.WORKSPACE}/bin:${goRoot}/bin:${env.PATH}",
+    "MAGEFILE_CACHE=${env.WORKSPACE}/.magefile",
+  ]) {
+    dir("${BEATS_DIR}") {
+      sh(label: "Create branch localVersion", script: "git checkout -b localVersion")
+      sh(label: "Install Go ${GO_VERSION}", script: ".ci/scripts/install-go.sh")
+      sh(label: "Install Mage", script: "make mage")
+    }
+    dir("${BASE_DIR}"){
+      git(credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba',
+        url:  "git@github.com:elastic/${env.REPO}.git")
+      sh(label: 'Update Beats script', script: """
+        git config --global user.email "you@example.com"
+        git config --global user.name "Your Name"
+        git config --global --add remote.origin.fetch "+refs/pull/*/head:refs/remotes/origin/pr/*"
+
+        mage pythonEnv
+        . build/ve/linux/bin/activate
+
+        go mod edit -replace github.com/elastic/beats/v7=\${GOPATH}/src/github.com/elastic/beats-local
+        git commit -m beats-update go.mod
+
+        make update
+        git commit -a -m "update"
+        echo "****************"
+        echo "****************"
+        echo "****************"
+        echo "****************"
+        go list -m -f {{.Path}} all | grep github.com/elastic/beats
+        echo "****************"
+        echo "****************"
+        echo "****************"
+        echo "****************"
+
+        make check
+        BEATS_MODULE=\$(go list -m -f {{.Path}} all | grep github.com/elastic/beats-local)
+        rsync -crv --delete \$(go list -m -f {{.Dir}} \$(BEATS_MODULE))/testing/environments testing/
```

Comment:
```suggestion
```

These files rarely change, and given that we're not testing the file contents I think we should skip this bit and just run `make check`

Reply:
removed